compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -442,6 +442,11 @@ private void checkXMLNamespacePrefixes(List<BLangXMLElementFilter> filters) {
         }
     }
 
+    private boolean isNonStructureJsonType(BType expType) {
+        return expType.tag == TypeTags.INT || expType.tag == TypeTags.FLOAT || expType.tag == TypeTags.DECIMAL
+                || expType.tag == TypeTags.NIL || expType.tag == TypeTags.STRING || expType.tag == TypeTags.JSON;
```

Comment:
Need to check int and string subtypes and boolean too, right? 

I think we can replace the method call with `org.wso2.ballerinalang.compiler.semantics.analyzer.Types#isSimpleBasicType`

```
types.isSimpleBasicType(type) || type.tag == TypeTags.JSON;
```

Also returning `true` when `expType.tag == TypeTags.JSON` where the method name is `isNonStructureJsonType` doesn't seem right to me.

Reply:
Remove the `isNonStructureJsonType` and use `isSimpleBasicType` function as mention.

Reply:
@MaryamZi According to spec-issue JSON cannot convert to boolean. Shall I create a new function for that or use `isSimpleBasicType` function as 
```
(expectedType.tag != TypeTags.BOOLEAN && types.isSimpleBasicType(expectedType.tag)) || expectedType.tag == TypeTags.JSON
```

Reply:
Fixed.